routers/web/repo/view.go

```diff
@@ -842,21 +842,19 @@ func renderDirectoryFiles(ctx *context.Context, timeout time.Duration) git.Entri
 	// Show latest commit info of repository in table header,
 	// or of directory if not in root directory.
 	ctx.Data["LatestCommit"] = latestCommit
-	if latestCommit != nil {
```

Comment:
Is it correct to do so?  If I understand correctly, `GetCommitsInfo` may return `([], nil, nil)`

Reply:
> Pretty sure this is correct as a empty repo will probably either not reach renderDirectoryFiles or bail out on GetCommitsInfo.

If it is the case, I think the code should return early if `latestCommit` is nil, and reports an error

Reply:
I agree I'm not certain that latestCommit is actually definitely not nil. I think in practice it's almost never nil but I don't think it's guaranteed.

Reply:
Yeah, probably better being safe than sorry in this case, especially around convoluted code like this.